rotkehlchen/chain/ethereum/modules/uniswap/utils.py

```diff
@@ -253,3 +270,302 @@ def historical_uniswap_v2_lp_price(
         token_price_func_args=[to_asset, timestamp],
         block_identifier=block_identifier,
     )
+
+
+def uniswap_v3_lp_token_balances(
+    userdb: 'DBHandler',
+    address: ChecksumEthAddress,
+    ethereum: 'EthereumManager',
+    premium: Optional[Premium],
+    known_assets: Set[EthereumToken],
+    unknown_assets: Set[EthereumToken],
+) -> List[NFTLiquidityPool]:
+    nft_manager_contract = EthereumContract(
+        address=UNISWAP_NFT_MANAGER_ADDRESS,
+        abi=UNISWAP_V3_NFT_MANAGER_ABI,
+        deployed_block=UNISWAP_V3_DEPLOYED_BLOCK,
+    )
+    address = to_checksum_address(address)
+    my_positions = nft_manager_contract.call(
+        ethereum=ethereum,
+        method_name="balanceOf",
+        arguments=[address],
+    )
+    balances: List[NFTLiquidityPool] = []
+    if my_positions <= 0:
+        return balances
+
+    chunks = list(get_chunks(list(range(my_positions)), n=10))
+    for chunk in chunks:
+        tokens_ids_multicall = multicall_2(
+            ethereum=ethereum,
+            require_success=True,
+            calls=[
+                (
+                    UNISWAP_NFT_MANAGER_ADDRESS,
+                    nft_manager_contract.encode('tokenOfOwnerByIndex', [address, index]),
+                )
+                for index in chunk
+            ],
+        )
+        tokens_ids = [
+            nft_manager_contract.decode(   # pylint: disable=unsubscriptable-object
+                result=data[1],
+                method_name='tokenOfOwnerByIndex',
+                arguments=[address, index],
+            )[0]
+            for index, data in enumerate(tokens_ids_multicall)
+        ]
```

Comment:
In every step I would be checking if the request of the multicall was successful or not since as your marked `require_success=True,` all the calls need to success. I would change it to `require_success=True` check the status of every result before keeping it for the next step and for those who failed I would log and error